## 9장 과제
컴퓨터공학부  
장수한(2021-14687)

## Problem 1

H0: 성별에 관계없이 완주시간은 동일하다.  
H1: 성별에 따라 완주시간이 다르다.  
$\alpha$=0.05

In [21]:
import pandas
# Cherry Blossom 10 mile run을 완주한 100명의 선수 데이터
run = pandas.read_table("./run10Samp.txt", sep=" ")
run


time  age gender state
1     88.31   59      M    MD
2    100.67   32      M    VA
3    109.52   33      F    VA
4     87.11   52      M    PA
5     73.02   40      M    DC
..      ...  ...    ...   ...
96    90.47   35      F    VA
97    94.32   28      F    VA
98    96.19   32      F    DC
99   104.31   32      F    MD
100   89.49   26      M    DC

[100 rows x 4 columns]

In [22]:
from statsmodels.formula.api import ols
import statsmodels.api as sm

model = ols('time ~ C(gender)', data=run).fit()
result_table = sm.stats.anova_lm(model, typ=2)
print(result_table)

                 sum_sq    df          F    PR(>F)
C(gender)   5196.208189   1.0  26.186817  0.000002
Residual   19445.983095  98.0        NaN       NaN


검정통계량 F의 값은 26.186817이며 p-value는 0.000002로 나타났다.  
p-value는 0.05보다 작다.  
따라서 귀무가설을 기각할 수 있다.   
즉, 유의수준 5%에서 성별에 따른 완주시간의 차이가 존재한다고 볼 충분한 통계적 근거가 있다.

## Problem 2

$\mu1$: 남자 모집단의 완주 시간에 대한 모평균  
$\mu2$: 여자 모집단의 완주 시간에 대한 모평균  

H0: $\mu1 = \mu2$  
H1: $\mu1 \neq \mu2$  

$\alpha$=0.05, 등분산을 가정한다.

In [23]:
from scipy.stats import ttest_ind
male = run[run['gender'] == 'M']['time']
female = run[run['gender'] == 'F']['time']
result = ttest_ind(male, female, equal_var=True)
print(result)

Ttest_indResult(statistic=-5.117305648181876, pvalue=1.5438819783437654e-06)


검정통계량의 값 F는 -5.117305648이다.  
p-value는 1.54e-06으로 0.05보다 작다.  
따라서 귀무가설을 기각할 수 있다.  
즉, 유의수준 5%에서 성별에 따라 완주 시간에 차이가 존재한다고 볼 충분한 통계적 근거가 있다.

## Problem 3

In [38]:
F, T = 26.186817, result.statistic
print('Problem 1에서의 검정통계량의 값: %f' % F)
print('Problem 2에서의 검정통계량의 값: %f' % T)
print('Problem 2에서의 검정통계량의 값의 제곱: %f\n' % (T**2))
print("F - T**2 = {}".format(F - T**2))


Problem 1에서의 검정통계량의 값: 26.186817
Problem 2에서의 검정통계량의 값: -5.117306
Problem 2에서의 검정통계량의 값의 제곱: 26.186817

F - T**2 = -9.691413183077202e-08


(2)의 검정통계량(T)의 값의 제곱은 (1)의 검정통계량(F)의 값과 같아지는 관계가 있다.  
남자, 여자의 명수를 n1, n2라 하자.
T는 t(n1+n2-2)를 따르고 F는 F(1, n1+n2-2)를 따른다.  
T분포와 F분포의 정의를 고려했을 때 T ~ t(k) 이면 T^2 ~ F(1, k)임과도 일맥상통한다.  

또 정성적으로 분석해보면 (1)과 (2) 모두 성별에 따른 완주 시간의 차이가 있는지 여부를 검정한 것으로  
남자, 여자의 완주 시간 모평균이 서로 차이가 있는지 확인하는 과정이라는 점에서 동일하다.  
(1)과 (2)의 유일한 차이점은 (2)의 검정통계량 T는 양측검정이 가능하고 (1)은 가능하지 않다는 점이다.  
따라서 P(F(1, n1+n2-2) >= F) = 2 * P( t(n1+n2-2) >= |T|) 를 만족해야하며 이는 F = T^2과 동치이다.  

## Problem 4

In [25]:
run['age'] = run['age'] // 10
run

time  age gender state
1     88.31    5      M    MD
2    100.67    3      M    VA
3    109.52    3      F    VA
4     87.11    5      M    PA
5     73.02    4      M    DC
..      ...  ...    ...   ...
96    90.47    3      F    VA
97    94.32    2      F    VA
98    96.19    3      F    DC
99   104.31    3      F    MD
100   89.49    2      M    DC

[100 rows x 4 columns]

In [37]:
print('연령대로 취할 수 있는 값:', set(run['age']))
print('연령대에 대한 인자 수준의 개수:', len(set(run['age'])))

연령대로 취할 수 있는 값: {2, 3, 4, 5, 6}
연령대에 대한 인자 수준의 개수: 5


$\alpha i$: '연령대' 인자의 i번째 수준의 효과  
$\beta i$: '성별' 인자의 i번째 수준의 효과  

H0: $\alpha 1 = \alpha 2 = \alpha 3 = \alpha 4 = \alpha 5 = 0$  
H1: Not H0  

H0: $\beta 1 = \beta 2 = 0$  
H1: Not H0

## Problem 5

In [30]:
model2 = ols('time ~ C(age) + C(gender)', data=run).fit()
res_table = sm.stats.anova_lm(model2, typ=2)
print(res_table)

                 sum_sq    df          F    PR(>F)
C(age)      1304.250566   4.0   1.689469  0.158923
C(gender)   5223.076786   1.0  27.062973  0.000001
Residual   18141.732528  94.0        NaN       NaN


1) 성별에 대한 검정  
검정통계량 F의 값은 27.062973이며 p-value는 0.000001이다.  
p-value가 0.05보다 작으므로 귀무가설을 기각할 수 있다.  
따라서 유의수준 5%에서 성별이 완주시간에 영향을 주었다고 볼 충분한 통계적 근거가 있다.  

2) 연령대에 대한 검정  
검정통계량 F의 값은 1.689469이며 p-value는 0.158923이다.  
p-value가 0.05보다 크므로 귀무가설을 기각할 수 없다.  
따라서 유의수준 5%에서 연령대가 완주시간에 영향을 주었다고 볼 통계적 근거가 없다.  

정리하자면 유의수준 5%에서 성별은 완주시간에 영향을 주고 연령대는 영향을 주지 않다고 볼 수 있다.